documentation/projects/proposals/publish_dataset/20230719-implementation_plan_initial_data_dump.md

```diff
@@ -0,0 +1,411 @@
+
+## Reviewers
+
+<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->
+
+- [ ] @obulat
+- [ ] @stacimc
+- [ ] @apolinario (optional)
+- [ ] @Skylion007 (optional)
+- [ ] @zackkrida (optional)
+
+## Project links
+
+<!-- Enumerate any references to other documents/pages, including milestones and other plans -->
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/2545)
+- [Project Proposal](https://docs.openverse.org/projects/proposals/publish_dataset/20230706-project_proposal.html)
+
+## Overview
+
+<!-- An overview of the implementation plan, if necessary. Save any specific steps for the section(s) below. -->
+
+This plan describes how the initial data dump of the media tables within
+Openverse's API database will occur on a technical level, as well as information
+regarding marketing and documentation of the initial data dump.
+[HuggingFace](https://huggingface.co/) has graciously offered to host the
+dataset for free, and this plan assumes that a handoff of our data to the
+HuggingFace team will be made to accomplish this.
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+Upon the completion of this implementation plan, the following should be
+available:
+
+- An unscheduled DAG which is capable of generating the data dump in S3.
+- A copy of the Openverse API database's primary media tables (i.e. `image` and
+  `audio`), in parquet format, is present within a "requester pays" S3 bucket
+  which can be accessed by the HuggingFace team for consumption into their
+  platform.
+- Documentation describing the "as-is" nature of the provided data, along with
+  descriptions of the fields present for each media type and the parquet files
+  themselves.
+- Updated internal documentation (e.g. API docs, our developer handbook, our
+  docs site, and potentially Openverse.org) referencing the dataset hosted by
+  HuggingFace (once it is made available)
+
+## Dependencies
+
+### Infrastructure
+
+<!-- Describe any infrastructure that will need to be provisioned or modified. In particular, identify associated potential cost changes. -->
+
+[ec2createoperator]:
+  https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/operators/ec2/index.html#airflow.providers.amazon.aws.operators.ec2.EC2CreateInstanceOperator
+[ec2terminateoperator]:
+  https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/operators/ec2/index.html#airflow.providers.amazon.aws.operators.ec2.EC2TerminateInstanceOperator
+
+**EC2 instance selection**
+
+Converting the data in our API database into any usable format will require a
+non-trivial amount of disk storage and compute time. Our desire is to deliver
```

Comment:
Also do not think it would take that much CPU either as it's more or less a straight table dump. Creating the parquet files may use a little bit of CPU, but it shouldn't saturate it unless you heavily parallelize the writing.